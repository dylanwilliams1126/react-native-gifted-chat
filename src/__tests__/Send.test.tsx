import 'react-native'
import React from 'react'
import renderer from 'react-test-renderer'

import { Send } from '../GiftedChat'

describe('Send', () => {
  it('should not render <Send /> and compare with snapshot', () => {
    const tree = renderer.create(<Send />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should always render <Send /> and compare with snapshot', () => {
    const tree = renderer.create(<Send alwaysShowSend />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should render <Send /> where there is input and compare with snapshot', () => {
    const tree = renderer.create(<Send text='test input' />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
