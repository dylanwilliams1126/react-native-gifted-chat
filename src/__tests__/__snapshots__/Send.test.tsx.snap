// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Send should always render <Send /> and compare with snapshot 1`] = `
<View
  accessibilityLabel="send"
  accessibilityRole="button"
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "height": 44,
      "justifyContent": "flex-end",
      "opacity": 1,
    }
  }
  testID="GC_SEND_TOUCHABLE"
>
  <View>
    <Text
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "color": "#0084ff",
            "fontSize": 17,
            "fontWeight": "600",
            "marginBottom": 12,
            "marginLeft": 10,
            "marginRight": 10,
          },
          undefined,
        ]
      }
    >
      Send
    </Text>
  </View>
</View>
`;

exports[`Send should not render <Send /> and compare with snapshot 1`] = `null`;

exports[`Send should render <Send /> where there is input and compare with snapshot 1`] = `
<View
  accessibilityLabel="send"
  accessibilityRole="button"
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "height": 44,
      "justifyContent": "flex-end",
      "opacity": 1,
    }
  }
  testID="GC_SEND_TOUCHABLE"
>
  <View>
    <Text
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "color": "#0084ff",
            "fontSize": 17,
            "fontWeight": "600",
            "marginBottom": 12,
            "marginLeft": 10,
            "marginRight": 10,
          },
          undefined,
        ]
      }
    >
      Send
    </Text>
  </View>
</View>
`;
